frame/lending/src/lib.rs

```diff
@@ -1249,102 +1351,14 @@ pub mod pallet {
 			account: &Self::AccountId,
 			amount: CollateralLpAmountOf<Self>,
 		) -> Result<(), DispatchError> {
-			ensure!(amount > Self::Balance::zero(), Error::<T>::CannotDepositZeroCollateral);
-			let market = Self::get_market(market_id)?;
-			let market_account = Self::account_id(market_id);
-
-			AccountCollateral::<T>::try_mutate(market_id, account, |collateral_balance| {
-				let new_collateral_balance =
-					collateral_balance.unwrap_or_default().safe_add(&amount)?;
-				collateral_balance.replace(new_collateral_balance);
-				Result::<(), DispatchError>::Ok(())
-			})?;
-
-			<T as Config>::MultiCurrency::transfer(
-				market.collateral_asset,
-				account,
-				&market_account,
-				amount,
-				true,
-			)?;
-			Ok(())
+			Self::do_deposit_collateral(market_id, account, amount.try_into_validated()?)
```

Comment:
idea of validated it validates during serde, so sure about this applicane. so may be ok